### PR TITLE
ISSUE-413 Implement webadmin task to reposition write rights for resources

### LINKED ISSUE #413: Write delegation for resource administrator
Upon adding resource administrator (update and creation)
Also position a write delegation onto the resource calendar for each administrator

Upon removing resource administrator (update and delete)
Also revoke this administrative right

Implements a route `POST resources?task=repositionWriteRights` that for each resource reset the write rights to the actual population of administrators (used eg for migration).

Definition of done: Resource administrator can update resource participation from OpenPaaS calendar SPA

Technically:

```
curl 'https://dav.linagora.com/calendars/{domainId}/{resourceId}.json' \
  -H 'Authorization: Bearer TECHNICAL_TOKEN' \
  -H 'Accept: application/json, text/plain, */*' \
  -H 'Content-Type: application/json;charset=UTF-8' \
  --data-raw '{"share":{"set":[{"dav:href":"mailto:admin@linagora.com","dav:read-write":true}],"remove":[]}}'

curl 'https://dav.linagora.com/calendars/5f50a663bdaffe002629099c/5f50a663bdaffe002629099c.json' \
  -H 'Authorization: Bearer TECHNICAL_TOKEN' \
  -H 'Accept: application/json, text/plain, */*' \
  -H 'Content-Type: application/json;charset=UTF-8' \
  --data-raw '{"share":{"set":[],"remove":[{"dav:href":"mailto:twake-calendar-dev@linagora.com"}]}}'
```

### PR DESCRIPTION
**Pull request by vttranlina**
resolve https://github.com/linagora/twake-calendar-side-service/issues/413

## Summary by CodeRabbit

* **New Features**
  * Web admin: POST /resources?task=repositionWriteRights to trigger a background task that repositions resource write rights; exposes task status, timestamps, and processed/failed counters.
  * New background task implementation to scan resources and reapply admin rights.

* **Tests**
  * Comprehensive integration tests covering success, partial failures, idempotence, cross-domain behavior, edge cases, and task lifecycle responses.

* **Documentation / DTO**
  * Added DTO-backed payload for task additional information (type, timestamp, counts).

### CODE REVIEW

File: calendar-webadmin/src/main/java/com/linagora/calendar/webadmin/task/RepositionResourceRightsTask.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        public Snapshot snapshot() {
+            return new Snapshot(processedResourceCount.get(), failedResourceCount.get());
+        }
+    }
+
+    public record Details(long processedResourceCount, long failedResourceCount, Instant timestamp) implements TaskExecutionDetails.AdditionalInformation {
+    }
+
+    public record RunningOptions(int resourcesPerSecond) {
+        public static final RunningOptions DEFAULT = new RunningOptions(1);
```
[the remark this answers is not in the document]
Comment (chibenwa) [reply calendar-webadmin/src/main/java/com/linagora/calendar/webadmin/task/RepositionResourceRightsTask.java]: I would prefer no throttling and a concatMap instead

Rationals: this 1 is likely intended as being sequential, 
In this use case we don't care about performance 
So let's use a better suited tool

File: calendar-webadmin/src/main/java/com/linagora/calendar/webadmin/task/RepositionResourceRightsTask.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    }
+
+    @Override
+    public Result run() {
+        return Flux.from(resourceDAO.findAll())
+            .filter(resource -> !resource.deleted())
+            .transform(ReactorUtils.<Resource, Task.Result>throttle()
+                .elements(runningOptions.resourcesPerSecond())
+                .per(Duration.ofSeconds(1))
+                .forOperation(resource -> reapplyWriteRights(resource, context)))
```
Comment (chibenwa) [inline calendar-webadmin/src/main/java/com/linagora/calendar/webadmin/task/RepositionResourceRightsTask.java]:
```suggestion
            .concatMap(resource -> reapplyWriteRights(resource, context))
```

**Review by hungphan227 — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]
